app/models/payment_gateway_setting.rb

```diff
@@ -18,9 +18,11 @@ class PaymentGatewaySetting < ApplicationRecord
   # In our DB :test is stored for some providers
 
   def symbolized_settings
-    keys = PaymentGateway::GATEWAYS.map(&:fields).map(&:keys).flatten.uniq
+    settings = gateway_settings
+    return settings.symbolize_keys if settings.is_a? Hash
```

Comment:
```suggestion
    return settings.symbolize_keys if settings.respond_to? :symbolize_keys
```
Better use duck typing here.